### DISCUSSION TITLE
Simulation divergence depends on stop_time/

### DISCUSSION THREAD

**Discussion by prevostthi** [OpenModelica/OpenModelica#12355] [General]
Hello,
I have a power system simulation that I wanted to run. The simulation went unstable at 7.52sec (with a stop_time = 10). To check what was going on, I changed stop_time to 7.508... run the simulation again... and it now becomes unstable at 5.6sec... If I change stop_time back to 10 sec, the simulation crashes at 7.52... I really can't understand what is going on... the interval is the same for both situations.
I attached the file that I try to run, together with the library needed to run it.
Any help would be more than welcome.
Thanks a lot
Thibault
[Dynawo_Modelica_Library_v1.4.1_test.zip](https://github.com/OpenModelica/OpenModelica/files/15165263/Dynawo_Modelica_Library_v1.4.1_test.zip)
[Test_linear.zip](https://github.com/OpenModelica/OpenModelica/files/15165298/Test_linear.zip)

**Comment by casella** [comment]
@prevostthi what do you mean as "unstable"?

**Reply by prevostthi** [reply]
By unstable, I mean that for example the power on line is suddenly diverging to 10^5 per unit. In fact most of the electrical values are going to extremely high values.

**Comment by lucas-mior** [comment]
@prevostthi I reproduced your issue, and I was able to fix it (at least the simulation finished successfully) by setting the maximum time step to a low value. Take a look into that. However, that library you got is very complex and you probably should debug it to know for sure what is going on. Also check your solver, I did the test using `dassl`.
![image](https://github.com/OpenModelica/OpenModelica/assets/49657055/9737b12b-7561-4f5c-a856-d7a6c3fd27db)

Also note that your library uses an older version of the Modelica standard library. You probably should update to version 4.

**Reply by prevostthi** [reply]
Thanks for your feedback, I was using Dassl too, I tried changing the maximum time step but it doesn't change anything. 
When you mentioned that the simulation finished successfully, what was the t_stop? In my case, the simulation still drive to crazy values for example for active power on lines. (P1pu).
Indeed the library is complex, but it is representing all the different assets and controls that are on an electrical transmission system, there is no solution to make is less complex.
Thanks for the tips about upgrading to version 4. I'll see how I can do that.

**Reply by casella** [reply]
I would suggest to simulate these systems with the compiler flag `--daeMode`. This avoids the transformation of the (spare) DAEs into (dense) explicit ODE form, and then uses the IDA algorithm to solve them.
